File: dart/sdk/example/quickstart.dart
import 'package:two_finance_sdk_client/two_finance_sdk_client.dart';

Future<void> main() async {
  final client = TwoFinanceClient.fromEnvironment();

  final indicators = await client.analytics.indicators();
  print('analytics indicators: $indicators');

  final plan = await client.planner.tradingPlan({
    'goal': 'prepare a BTC rebalancing plan',
    'useAnalytics': true,
    'useTrading': true,
  });
  print('planner response: $plan');
}

File: cpp/examples/quickstart.cpp
#include "twofinance/sdk_client.hpp"

#include <iostream>

int main() {
  twofinance::SdkConfig config = twofinance::config_from_environment();

  twofinance::Transport transport = [](const twofinance::HttpRequest& request) {
    std::cout << request.method << " " << request.url << "\n";
    return twofinance::HttpResponse{200, "{\"ok\":true}"};
  };

  twofinance::SdkClient client(config, transport);
  client.analytics.indicators();
  client.planner.trading_plan(
      "{\"goal\":\"prepare a BTC rebalancing plan\",\"useAnalytics\":true,\"useTrading\":true}",
      true,
      true);

  return 0;
}
